src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableList_1.Node.cs

```diff
@@ -26,7 +26,7 @@ internal sealed class Node : IBinaryTree<T>, IEnumerable<T>
             /// <summary>
             /// The key associated with this node.
             /// </summary>
-            private T _key;
+            private T _key = default!;
```

Comment:
Why did you choose to do it this way?  A comment would be helpful.

Reply:
Empty constructor complaining it is not set which only used for creating `EmptyNode`. Didn't make it `[MaybeNull]` as it is set for all other normal nodes and shouldn't be null. Key for `EmptyNode` is not used and apparently shouldn't be used.